From ff72078491c0276856e348a83edfed14fa27975f Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Thu, 8 Feb 2018 14:47:54 +0800
Subject: [PATCH] arm64: dts: rockchip: Fix some adc key issues for rk3326/px30
 board

1. add vref voltage 1.8v;
2. Alphabetical order;
3. Fix the press-threshold-microvolt;
4. Add poll interval 100ms

Change-Id: If5e29f5073155be3022053cc3516dda88c5db223
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 .../boot/dts/rockchip/px30-evb-ddr3-v10.dts   | 34 ++++++++++---------
 .../boot/dts/rockchip/px30-evb-ddr4-v10.dts   | 22 ++++++------
 .../boot/dts/rockchip/rk3326-863-lp3-v10.dts  | 16 +++++----
 .../boot/dts/rockchip/rk3326-evb-lp3-v10.dts  | 34 ++++++++++---------
 4 files changed, 57 insertions(+), 49 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
index 97e65dbe75d7..f928d78a4c5b 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
@@ -21,36 +21,37 @@
 		compatible = "adc-keys";
 		io-channels = <&saradc 2>;
 		io-channel-names = "buttons";
+		poll-interval = <100>;
 		keyup-threshold-microvolt = <1800000>;
 
-		vol-up-key {
-			linux,code = <KEY_VOLUMEUP>;
-			label = "volume up";
-			press-threshold-microvolt = <10000>;
-		};
-
-		vol-down-key {
-			linux,code = <KEY_VOLUMEDOWN>;
-			label = "volume down";
-			press-threshold-microvolt = <170000>;
+		esc-key {
+			linux,code = <KEY_ESC>;
+			label = "esc";
+			press-threshold-microvolt = <1270000>;
 		};
 
 		home-key {
 			linux,code = <KEY_HOME>;
 			label = "home";
-			press-threshold-microvolt = <254000>;
+			press-threshold-microvolt = <602000>;
 		};
 
 		menu-key {
 			linux,code = <KEY_MENU>;
 			label = "menu";
-			press-threshold-microvolt = <414000>;
+			press-threshold-microvolt = <952000>;
 		};
 
-		esc-key {
-			linux,code = <KEY_ESC>;
-			label = "esc";
-			press-threshold-microvolt = <614000>;
+		vol-down-key {
+			linux,code = <KEY_VOLUMEDOWN>;
+			label = "volume down";
+			press-threshold-microvolt = <290000>;
+		};
+
+		vol-up-key {
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <17000>;
 		};
 	};
 
@@ -587,6 +588,7 @@
 
 &saradc {
 	status = "okay";
+	vref-supply = <&vcc1v8_soc>;
 };
 
 &sdmmc {
diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
index 9f51bef0ac96..21bfae817f48 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
@@ -21,36 +21,37 @@
 		compatible = "adc-keys";
 		io-channels = <&saradc 2>;
 		io-channel-names = "buttons";
+		poll-interval = <100>;
 		keyup-threshold-microvolt = <1800000>;
 
 		esc-key {
 			linux,code = <KEY_ESC>;
 			label = "esc";
-			press-threshold-microvolt = <614000>;
+			press-threshold-microvolt = <1270000>;
 		};
 
 		home-key {
 			linux,code = <KEY_HOME>;
 			label = "home";
-			press-threshold-microvolt = <254000>;
+			press-threshold-microvolt = <602000>;
 		};
 
 		menu-key {
 			linux,code = <KEY_MENU>;
 			label = "menu";
-			press-threshold-microvolt = <414000>;
-		};
-
-		vol-up-key {
-			linux,code = <KEY_VOLUMEUP>;
-			label = "volume up";
-			press-threshold-microvolt = <10000>;
+			press-threshold-microvolt = <952000>;
 		};
 
 		vol-down-key {
 			linux,code = <KEY_VOLUMEDOWN>;
 			label = "volume down";
-			press-threshold-microvolt = <170000>;
+			press-threshold-microvolt = <290000>;
+		};
+
+		vol-up-key {
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <17000>;
 		};
 	};
 
@@ -587,6 +588,7 @@
 
 &saradc {
 	status = "okay";
+	vref-supply = <&vcc1v8_soc>;
 };
 
 &sdmmc {
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
index dee77de7ebbd..648149d24450 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
@@ -22,18 +22,19 @@
 		compatible = "adc-keys";
 		io-channels = <&saradc 2>;
 		io-channel-names = "buttons";
+		poll-interval = <100>;
 		keyup-threshold-microvolt = <1800000>;
 
-		vol-up-key {
-			linux,code = <KEY_VOLUMEUP>;
-			label = "volume up";
-			press-threshold-microvolt = <10000>;
-		};
-
 		vol-down-key {
 			linux,code = <KEY_VOLUMEDOWN>;
 			label = "volume down";
-			press-threshold-microvolt = <170000>;
+			press-threshold-microvolt = <290000>;
+		};
+
+		vol-up-key {
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <17000>;
 		};
 	};
 
@@ -539,6 +540,7 @@
 
 &saradc {
 	status = "okay";
+	vref-supply = <&vcc1v8_soc>;
 };
 
 &sdmmc {
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
index 1d3b99554550..b24b015c10b4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
@@ -22,36 +22,37 @@
 		compatible = "adc-keys";
 		io-channels = <&saradc 2>;
 		io-channel-names = "buttons";
+		poll-interval = <100>;
 		keyup-threshold-microvolt = <1800000>;
 
-		vol-up-key {
-			linux,code = <KEY_VOLUMEUP>;
-			label = "volume up";
-			press-threshold-microvolt = <10000>;
-		};
-
-		vol-down-key {
-			linux,code = <KEY_VOLUMEDOWN>;
-			label = "volume down";
-			press-threshold-microvolt = <170000>;
+		esc-key {
+			linux,code = <KEY_ESC>;
+			label = "esc";
+			press-threshold-microvolt = <1270000>;
 		};
 
 		home-key {
 			linux,code = <KEY_HOME>;
 			label = "home";
-			press-threshold-microvolt = <254000>;
+			press-threshold-microvolt = <602000>;
 		};
 
 		menu-key {
 			linux,code = <KEY_MENU>;
 			label = "menu";
-			press-threshold-microvolt = <414000>;
+			press-threshold-microvolt = <952000>;
 		};
 
-		esc-key {
-			linux,code = <KEY_ESC>;
-			label = "esc";
-			press-threshold-microvolt = <614000>;
+		vol-down-key {
+			linux,code = <KEY_VOLUMEDOWN>;
+			label = "volume down";
+			press-threshold-microvolt = <290000>;
+		};
+
+		vol-up-key {
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <17000>;
 		};
 	};
 
@@ -665,6 +666,7 @@
 
 &saradc {
 	status = "okay";
+	vref-supply = <&vcc1v8_soc>;
 };
 
 &sdmmc {
-- 
2.35.3

